src/compiler/output-targets/dist-custom-elements/index.ts

```diff
@@ -38,38 +49,66 @@ export const outputCustomElements = async (
   const bundlingEventMessage = 'generate custom elements';
   const timespan = buildCtx.createTimeSpan(`${bundlingEventMessage} started`);
 
-  await Promise.all(outputTargets.map((o) => bundleCustomElements(config, compilerCtx, buildCtx, o)));
+  await Promise.all(outputTargets.map((target) => bundleCustomElements(config, compilerCtx, buildCtx, target)));
 
   timespan.finish(`${bundlingEventMessage} finished`);
 };
 
-const bundleCustomElements = async (
+/**
+ * Get bundle options for our current build and compiler context which we'll use
+ * to generate a Rollup build and so on.
+ *
+ * @param config user-supplied Stencil configuration
+ * @param buildCtx the current build context
+ * @param compilerCtx the current compiler context
+ * @param outputTarget the outputTarget we're currently dealing with
+ * @returns bundle options suitable for generating a rollup configuration
+ */
+export const getBundleOptions = (
+  config: d.Config,
+  buildCtx: d.BuildCtx,
+  compilerCtx: d.CompilerCtx,
+  outputTarget: d.OutputTargetDistCustomElements
+): BundleOptions => {
+  const bundleOpts: BundleOptions = {
+    id: 'customElements',
+    platform: 'client',
+    conditionals: getCustomElementsBuildConditionals(config, buildCtx.components),
+    customTransformers: getCustomElementCustomTransformer(config, compilerCtx, buildCtx.components, outputTarget),
+    externalRuntime: !!outputTarget.externalRuntime,
+    inlineWorkers: true,
+    inputs: {
+      // Here we prefix our index chunk with '\0' to tell Rollup that we're
+      // going to be using virtual modules with this module. A leading '\0'
+      // prevents other plugins from messing with the module. We generate a
+      // string for the index chunk below in the `loader` property.
+      //
+      // @see {@link https://rollupjs.org/guide/en/#conventions} for more info.
```

Comment:
this and other similar comments are my best attempt at understanding what's going on - it's entirely possible that I misunderstood something though!

Reply:
Indeed it is not!